scripts/metric.py

```diff
@@ -6,63 +6,131 @@
+    and custom metrics like Spearman correlation. Add new custom metrics here.
+
+    Args:
+        actual_values: List of actual/ground truth values
+        predicted_values: List of predicted values
+
+    Returns:
+        Dictionary of metric names and their values
+    """
+    # Calculate confusion matrix metrics
+    cm = ConfusionMatrix(
+        actual_vector=actual_values,
+        predict_vector=predicted_values,
+    )
+
+    all_metrics = dict(cm.overall_stat.items())
+
+    # Calculate custom metrics
+    # Add your custom metrics below this line
+
+    # Custom metric: Average Spearman correlation
+    spearman_corr, _ = spearmanr(actual_values, predicted_values)
+    all_metrics["Average Spearman"] = spearman_corr
+
+    return all_metrics
 
 
 def calc(
-    output: Path, metric: Path, actual_vector_col: str, predict_vector_col: str
+    prediction: Path,
+    metric: Path,
+    plot: Path,
+    actual_vector_col: str,
+    predict_vector_col: str,
+    selected_metrics: list[str] | None = None,
 ) -> Path:
-    """Calculate performance metrics from prediction output and save to CSV.
+    """Calculate performance metrics from prediction output and save to metric and plot JSON formats.
 
     Reads prediction results from a CSV file, computes classification metrics using
-    a confusion matrix. All metrics are saved to a CSV file.
+    a confusion matrix. All metrics are saved to a JSON file, and a plot-ready version
+    is saved to a separate JSON file for visualization purposes.
 
     Args:
-        output: Path to the CSV file containing prediction results
-        metric: Path where the calculated metrics CSV will be saved
+        prediction: Path to the CSV file containing prediction results
+        metric: Path where the calculated metrics JSON will be saved
+        plot: Path where the plot-ready metrics JSON will be saved
         actual_vector_col: Column name containing actual/ground truth values
         predict_vector_col: Column name containing predicted values
+        selected_metrics: Optional list of metric names to include. If None, all metrics are included.
+
+    Returns:
+        Tuple of (metric, plot) paths where the files were saved
     """
 
     print("Start to calculate metrics.")
 
-    output_dataframe = pl.read_csv(output)
+    prediction_dataframe = pl.read_csv(prediction).drop_nulls()
 
-    cm = ConfusionMatrix(
-        actual_vector=output_dataframe[actual_vector_col].to_list(),
-        predict_vector=output_dataframe[predict_vector_col].to_list(),
-    )
+    actual_values = prediction_dataframe[actual_vector_col].to_list()
+    predicted_values = prediction_dataframe[predict_vector_col].to_list()
+    
+    all_metrics = calculate_all_metrics(actual_values, predicted_values)
 
-    metrics_data = [
-        {"metric_name": key, "metric_value": str(value)}
-        for key, value in cm.overall_stat.items()
-    ]
+    if selected_metrics:
```

Comment:
For efficiency, you would like to only calculate the selected metrics, now all metrics are calculated still when a selection is provided. (Not a big deal but prettier)